libs/sherlock/src/lib/reactor/rxjs-interop.tests.ts

```diff
@@ -0,0 +1,313 @@
+                        expect(values).toEqual(['a']);
+                        a$.set('b');
+                        expect(values).toEqual(['a', 'b']);
+                    });
+
+                    test('until Error', () => {
+                        a$.setError('some error');
+                        a$.set('c');
+                        expect(values).toEqual(['a', 'b']);
+                    });
+
+                    test('until Final', () => {
+                        a$.setFinal('c');
+                        expect(values).toEqual(['a', 'b', 'c']);
+                    });
+
+                    test('until unsubscribe', () => {
+                        unsubscribe();
+                        a$.set('c');
+                        expect(values).toEqual(['a', 'b']);
+                    });
+                });
+
+                describe('with only an error callback', () => {
+                    let errorCb: jest.Mock, unsubscribe: () => void;
+                    beforeEach(() => {
+                        ({ unsubscribe } = subscribe(undefined, (errorCb = jest.fn())));
+                        a$.set('b');
+                    });
+
+                    test('until Error', () => {
+                        a$.setError('some error');
+                        a$.set('c');
+                        a$.setError('some other error');
+                        expect(errorCb).toHaveBeenCalledTimes(1);
+                        expect(errorCb).toHaveBeenCalledWith('some error');
+                    });
+
+                    test('until Final', () => {
+                        a$.setFinal('c');
+                        expect(errorCb).not.toHaveBeenCalled();
+                    });
+
+                    test('until unsubscribe', () => {
+                        unsubscribe();
+                        a$.set('c');
+                        expect(errorCb).not.toHaveBeenCalled();
+                    });
+                });
+
+                describe('with only a complete callback', () => {
+                    let completeCb: jest.Mock, unsubscribe: () => void;
+                    beforeEach(() => {
+                        ({ unsubscribe } = subscribe(undefined, undefined, (completeCb = jest.fn())));
+                        a$.set('b');
+                    });
+
+                    test('until Error', () => {
+                        a$.setError('some error');
+                        a$.set('c');
+                        a$.setError('some other error');
+                        expect(completeCb).not.toHaveBeenCalled();
+                    });
+
+                    test('until Final', () => {
+                        a$.setFinal('c');
+                        expect(completeCb).toHaveBeenCalledTimes(1);
+                    });
+
+                    test('until unsubscribe', () => {
+                        unsubscribe();
+                        a$.set('c');
```

Comment:
The real proof of the pudding would be to `.setFinal()` here. Since `.set()` wouldn't have triggered `completeCb` anyway.
```suggestion
                        a$.setFinal('c');
```